src/transform/src/lib.rs

```diff
@@ -340,17 +340,22 @@ impl Optimizer {
         Self { transforms }
     }
 
-    /// Optimizes the supplied relation expression returning an optimized relation.
+    /// Optimizes the supplied relation expression.
+    ///
+    /// These optimizations are performed with no information about available arrangements,
+    /// which makes them suitable for pre-optimization before dataflow deployment.
     pub fn optimize(
         &mut self,
         mut relation: MirRelationExpr,
-        indexes: &HashMap<GlobalId, Vec<(GlobalId, Vec<MirScalarExpr>)>>,
     ) -> Result<expr::OptimizedMirRelationExpr, TransformError> {
-        self.transform(&mut relation, indexes)?;
+        self.transform(&mut relation, &HashMap::new())?;
```

Comment:
It looks like the only leftover call to `transform` is from `dataflow.rs`, which includes indexes. Is that supposed to be the case?

Reply:
Yes, that is the one place we actually mean to invoke the method with specific indexes (as part of the last optimization before building the dataflow, as a function of the indexes that are imported).